src/util/scoped_ctrl_c.cpp

```diff
@@ -43,15 +42,17 @@ scoped_ctrl_c::scoped_ctrl_c(event_handler & eh, bool once, bool enabled):
     m_old_scoped_ctrl_c(g_obj) {
     if (m_enabled) {
         g_obj = this;
-        m_old_handler = signal(SIGINT, on_ctrl_c); 
+        struct sigaction sa;
+        sa.sa_handler = on_ctrl_c;
+        sa.sa_flags = SA_ONSTACK;
```

Comment:
but without calling `sigaltstack`, isn't this useless?

My man page says:

> Call the signal handler on an alternate signal stack
              provided by [sigaltstack(2)](https://www.man7.org/linux/man-pages/man2/sigaltstack.2.html).  If an alternate stack is not
              available, the default stack will be used.  This flag is
              meaningful only when establishing a signal handler.

Reply:
It is what Github WorkSpaces provided in the first shot. 
Besides looking suspicious, it also isn't cross-platform. For Windows, we would not use this.
Then also, I am not sure if the code in 'shell' is broken because it only considers sigint.
Altogether more than a quick change, but something that you as the highest esteemed and glorious expert in operating sophisticated machinery may find a spare moment to consider when not occupied near sunny grass lawns or ocean side camps (it is August, isn't it?).